frontend/device/generic/device.lua

```diff
@@ -116,8 +117,14 @@ end
 -- ONLY used for Kobo and PocketBook devices
 function Device:onPowerEvent(ev)
     local Screensaver = require("ui/screensaver")
+    local network_manager = require("ui/network/manager")
     if (ev == "Power" or ev == "Suspend") and not self.screen_saver_mode then
         self.powerd:beforeSuspend()
+        self.wifi_was_on = network_manager:getWifiStatus()
```

Comment:
I think that won't work. If this feature is on, users may not even click on network menu anymore, and both getWifiStatus and promptWifiOn won't be called.